Rebrand candidate files

Jaifroid opened the pull request:
Implements #262

mossroy commented on manifest.json:
```diff
@@ -29,7 +29,7 @@
     "applications": {
         "gecko": {
-            "id": "kiwix-html5-unlisted@kiwix.org"
+            "id": "kiwix-js-unlisted@kiwix.org"
```
At least for now, we need to keep this id untouched, else Firefox will probably consider the new versions as a different addon, and will probably won't do the upgrade for end-users

mossroy commented on scripts/package_firefox_extension.sh:
```diff
@@ -53,7 +53,7 @@ if [ "${TAG}zz" == "zz" ]; then
     # When packaging a public version, we need to prepare a 'listed' extension package to submit to Mozilla
     echo "Replacing the Firefox 'unlisted' extension id by the 'listed' one to be accepted by Mozilla"
-    sed -i -e "s/kiwix-html5-unlisted@kiwix.org/kiwix-html5-listed@kiwix.org/" manifest.json
+    sed -i -e "s/kiwix-js-unlisted@kiwix.org/kiwix-js-listed@kiwix.org/" manifest.json
```
This needs to be kept as it is for now (see previous comment on the Firefox addon id)

mossroy commented on scripts/package_chrome_extension.sh:
```diff
@@ -18,7 +18,7 @@ cd ..
     # Package the extension with Chromium, if we're not packaging a public version
     echo "Signing the extension for Chrome with a local Chromium, version $VERSION"
-    chromium-browser --no-sandbox --pack-extension=tmp --pack-extension-key=./scripts/kiwix-html5.pem
+    chromium-browser --no-sandbox --pack-extension=tmp --pack-extension-key=./scripts/kiwix-js.pem
```
This one will be more complicated to change, because it involves a "secret" file (kiwix-html5.pem), which is not commited, and included (for Travis) in an encrypted form inside secret_files.tar.gz.enc.

mossroy commented on README.md:
```diff
@@ -1,4 +1,4 @@
-kiwix-html5
+kiwix-JS
```
Maybe with an uppercase on the K, to follow https://github.com/kiwix/kiwix-html5/issues/248#issuecomment-309490013

Jaifroid commented:
OK, I've reverted those changes. @mossroy please double check.

mossroy commented on scripts/package_firefox_extension.sh:
```diff
@@ -53,7 +53,7 @@ if [ "${TAG}zz" == "zz" ]; then
     # When packaging a public version, we need to prepare a 'listed' extension package to submit to Mozilla
     echo "Replacing the Firefox 'unlisted' extension id by the 'listed' one to be accepted by Mozilla"
-    sed -i -e "s/kiwix-html5-unlisted@kiwix.org/kiwix-html5-listed@kiwix.org/" manifest.json
+    sed -i -e "s/kiwix-html5-unlisted@kiwix.org/kiwix-js-listed@kiwix.org/" manifest.json
```
This should not be modified either, as "kiwix-html5-listed@kiwix.org" is the id known by Mozilla to distribute the addon

Jaifroid replied:
Done!

mossroy commented:
This looks good to me.
One extra "HTML5" string can be replaced by "JS" in the subject of the mailto: link, line 137 of index.html, but it's minor.
In any case, I think this PR is a great candidate for you to practice squashing all the commits into a single one :-)

Jaifroid commented:
Well, I did the "rebase -i master" command on my local machine, changed the commits to "f" (other than the first, which I kept as "p"), saved, got the response "Successfully rebased and updated refs/heads/Rebrand-candidate-files", then pushed to the server. But it doesn't look any different here. How do I know that the rebase and squash worked? Is it just that some of the earlier commits now have "Show outdated" next to them? Also, is there any advantage in squashing locally rather than using the "squash and merge" command here (down arrow next to "Merge pull request" button)?

mossroy commented:
The file changes are OK for me, but the commits are not squashed.
If they were, the PR on github would display the same file changes, but in only one commit.
The "outdated" parts are the discussions we had on parts of the code that have been modified since then. They are outdated because you did the changes I asked.

Locally on your computer, if you type "git log", do you still see the "revert" commits you did?
If you don't see them any more, it's only a matter of pushing. Try again to push with "--force" option, which is required after rebase or squash.
If you still see them, it means you did not squash properly on your computer (but the steps you mentioned seem ok to me).

Doing the squash from github would work in this case. It means the squash will only be done on github, not in your local branch (which should not be an issue in this case).
If you don't manage to do the squash, I can do that if you prefer, no problem.

Doing the squash locally have a few advantages in my opinion : 
- it separates squashing (which should be done by the developer, who chooses the right way to "merge" the commit messages) and merging (which should be done by the reviewer)
- it enables squashing and rebasing at the same time (it does not seem to be possible to do both in github interface)
- it makes people learn a way that will work with any git repo (not necessarily github)

Jaifroid commented:
I think I must have messed something up. They're squashed on my local machines, but "git push --force Rebrand-candidate files" results in:

```
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push Rebrand-candidate-files HEAD:<name-of-remote-branch>

```
The suggested command (with substituted branch name) results in:

```
fatal: 'Rebrand-candidate-files' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I can't retrace what I did the first time I attempted this, and clearly my local repository is in a strange state now. So to avoid doing something catastrophic, I'm going to squash and merge these changes, and re-clone my repo in a clean state! Will try again next time... Thanks for your help.

mossroy commented:
All right, no problem

mossroy commented:
You also can delete the branch, with the button github provides here

Jaifroid commented:
Ah yes, sorry I forgot. Done now.
